### PR TITLE
Week 5 

### PR DESCRIPTION
**Pull request by melissa-perez**
- Create MongoDB
- Create Controllers/Routes/Model

### CODE REVIEW

File: 02-express-tutorial/app.js [hunk 1]
```diff
@@ -1 +1,73 @@
...
+
+// optional
+app.get("/test", auth, (req, res)=> {
+    return res.status(200).json({message: `Welcome, ${req.user}! `})
+})
+
+app.post("/logon", (req, res) => {
+    const {name} = req.body
+
+    if(name){
```
Comment (jonathan-david-johnson) [inline 02-express-tutorial/app.js]: Here you're checking for if (name) and doing the happy path of the code inside the if. Then you're handling the error case after the if.

This is absolutely fine and many programmers work this way. But I prefer to handle errors at the top and then have the happy path logic below and not inside a code block.

So I'd have it
```
if (errorcase) { 
  handleError...
  }
  
  ... happy path code
  }
```

I find this easier for my brain.

File: 03-task-manager/optionalArrayMethodsReviewExtraAssignment.js [hunk 2]
```diff
@@ -184,7 +184,7 @@ console.log('everyone last name', everyonesLastName);
 
 //////// CHALLENGE: Filter to the people who followed the right
 // "right format" is "<first name> <last name>" with a single space!
-const rightFormat = /^\w+ \w+$/;
+const rightFormat = /\w^+ \w+$/;
```
Comment (jonathan-david-johnson) [inline 03-task-manager/optionalArrayMethodsReviewExtraAssignment.js]: Why was this line changed to this?
This regex is invalid.

**Review by jonathan-david-johnson — APPROVED** [review]
I submitted one comment on the optional assignment. 
Otherwise this looks really good.

**Review by jonathan-david-johnson — APPROVED** [review]

File: 03-task-manager/optionalArrayMethodsReviewExtraAssignment.js (hunk 2 above)
Comment (melissa-perez) [reply 03-task-manager/optionalArrayMethodsReviewExtraAssignment.js]: I don't know actually. I noticed a file change but I don't remember doing any line updates in the optional file. I'll change back!
